### PR TITLE
After service is removed from consul it should be removed from Servic…

### PR DESCRIPTION
**Pull request by Automaat**
…esState

### CODE REVIEW

File: envoy-control-services/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/services/ServicesState.kt [hunk 1]
```diff
@@ -3,30 +3,39 @@ package pl.allegro.tech.servicemesh.envoycontrol.services
...
     fun hasService(serviceName: String): Boolean = serviceNameToInstances.containsKey(serviceName)
     fun serviceNames(): Set<ServiceName> = serviceNameToInstances.keys
     fun allInstances(): Collection<ServiceInstances> = serviceNameToInstances.values
 
-    fun remove(serviceName: ServiceName): ServicesState {
-        // TODO: https://github.com/allegro/envoy-control/issues/11
-        return change(ServiceInstances(serviceName, instances = emptySet()))
+
+    fun remove(serviceName: ServiceName): Boolean {
+        return if (serviceName in serviceNameToInstances) {
```
Comment (gawarz) [inline envoy-control-services/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/services/ServicesState.kt:16]: Method remove returns previous value or null. You dont need to use if.
Please consider:
`serviceNameToInstances.remove(serviceName) != null`

Comment (Automaat) [reply envoy-control-services/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/services/ServicesState.kt]: thanks for pointing it out. Fixed

File: envoy-control-services/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/services/ServicesState.kt [hunk 2]
```diff
@@ -3,30 +3,32 @@ package pl.allegro.tech.servicemesh.envoycontrol.services
 typealias ServiceName = String
 
 data class ServicesState(
-    val serviceNameToInstances: Map<ServiceName, ServiceInstances> = emptyMap()
+    val serviceNameToInstances: MutableMap<ServiceName, ServiceInstances> = mutableMapOf()
```
Comment (gawarz) [inline envoy-control-services/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/services/ServicesState.kt:6]: This class has a lack of encapsulation. This wasn't big deal before but now `serviceNameToInstances` is mutable. Please consider to encapsulate it with unmutableMap interface

Comment (Automaat) [reply envoy-control-services/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/services/ServicesState.kt]: done

**Review by KSmigielski — APPROVED** [review]

**Review by gawarz — APPROVED** [review]

**Pull request merged**